tests/gtest/avifallocationtest.cc

```diff
@@ -58,22 +58,17 @@ TEST(AllocationTest, MinimumValidDimensions) {
   TestAllocation(1, 1, 8, AVIF_RESULT_OK);
 }
 
-TEST(AllocationTest, MaximumValidDimensions) {
-  // On 32-bit builds, malloc() will fail with fairly low sizes.
-  // Adapt the tests to take that into account.
-  constexpr bool kIsPlatform64b = sizeof(void*) > 4;
-  constexpr uint32_t kMaxAllocatableDimension =
-      kIsPlatform64b ? std::numeric_limits<typeof(avifImage::width)>::max()
-                     : 134217728;  // Up to 1 GB total for YUVA
-
-  // 8 bits
-  TestAllocation(kMaxAllocatableDimension, 1, 8, AVIF_RESULT_OK);
-  TestAllocation(1, kMaxAllocatableDimension, 8, AVIF_RESULT_OK);
-  // 12 bits (impacts the width because avifImage stride is stored as uint32_t)
-  TestAllocation(kMaxAllocatableDimension / 2, 1, 12, AVIF_RESULT_OK);
-  TestAllocation(1, kMaxAllocatableDimension, 12, AVIF_RESULT_OK);
-  // Some high number of bytes that malloc() accepts to allocate.
-  TestAllocation(1024 * 16, 1024 * 8, 12, AVIF_RESULT_OK);  // Up to 1 GB total
+// The libavif API accepts allocating up to SIZE_MAX bytes. Testing this
+// threshold is unrealistic so allocate 1 GB: that should pass on all platforms
```

Comment:
Nit: The comment "The libavif API accepts allocating up to SIZE_MAX bytes." is a little confusing because it seems to imply that up to `SIZE_MAX` bytes can be allocated. I suggest rephrasing the comment, for example, "Up to SIZE_MAX can be passed as the input argument to avifAlloc()."

It is also confusing to refer to `SIZE_MAX` bytes as a threshold. `SIZE_MAX` is the maximum value of the `size_t` type (the type of the input parameter of `avifAlloc()`). `SIZE_MAX` is not a threshold for memory allocation (i.e., the maximum number of bytes that can be allocated).

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/1015